lib/fluent/plugin/in_tail.rb

```diff
@@ -488,7 +488,7 @@ def stop_watchers(targets_info, immediate: false, unwatched: false, remove_watch
     end
 
     def close_watcher_handles
-      @tails.keys.each do |path|
+      @tails.each_key do |path|
         tw = @tails.delete(path)
```

Comment:
I think we should not use `each_key` and so on for blocks that make such destructive changes.
I know this works without a problem, but I think this is unfavorable.

I would like to hear from other maintainers.

Reply:
Because I'm not so familiar with Ruby, I'm not sure it's usual usage or not.
But I feel uncomfortable too. At least Ruby doesn't allow adding a key while `each_key`:
```ruby
irb(main):018:1* a.each_key do |key|
irb(main):019:1*   a[key.to_s + "1"] = a[key] + 1
irb(main):020:0> end
(irb):19:in `block in <main>': can't add a new key into hash during iteration (RuntimeError)
	from (irb):18:in `each_key'
	from (irb):18:in `<main>'
	from /usr/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /usr/bin/irb:23:in `load'
	from /usr/bin/irb:23:in `<main>'
```